dev/devicelab/bin/tasks/flutter_run_test.dart

```diff
@@ -0,0 +1,68 @@
+// Copyright 2018 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:io';
+
+import 'package:path/path.dart' as path;
+
+import 'package:flutter_devicelab/framework/adb.dart';
+import 'package:flutter_devicelab/framework/framework.dart';
+import 'package:flutter_devicelab/framework/utils.dart';
+
+final Directory flutterGalleryDir = dir(path.join(flutterDirectory.path, 'examples/hello_world'));
+final File runTestSource = File(path.join(
+  flutterDirectory.path, 'dev', 'automated_tests', 'flutter_run_test', 'flutter_run_test.dart',
+));
+const Pattern passedMessageMatch = '+0: example passed';
+const Pattern failedMessageMatch = '+1: example failed [E]';
+const Pattern skippedMessageMatch = '+1 -1: example skipped';
+const Pattern finishedMessageMatch = '+1 ~1 -1: Some tests failed.';
+
+Future<void> main() async {
+  deviceOperatingSystem = DeviceOperatingSystem.android;
+  await task(createFlutterRunTask());
+}
+
+TaskFunction createFlutterRunTask() {
+  return () async {
```

Comment:
Do you need an inner closure? Why not put the logic directly inside the top-level function then tear it off?

Reply:
i.e. `await task(flutterRunTask);`

Reply:
A result of some over-eager copy pasting, fixed